source/common/stats/thread_local_store.cc

```diff
@@ -215,18 +261,22 @@ Histogram& ThreadLocalStoreImpl::ScopeImpl::histogram(const std::string& name) {
 
   std::unique_lock<std::mutex> lock(parent_.lock_);
   HistogramSharedPtr& central_ref = central_cache_.histograms_[final_name];
+
+  std::vector<Tag> tags;
+  std::string tag_extracted_name = parent_.getTagsForName(final_name, tags);
   if (!central_ref) {
-    std::vector<Tag> tags;
-    std::string tag_extracted_name = parent_.getTagsForName(final_name, tags);
-    central_ref.reset(
-        new HistogramImpl(final_name, parent_, std::move(tag_extracted_name), std::move(tags)));
+    central_ref.reset(new HistogramParentImpl(final_name, parent_, std::move(tag_extracted_name),
+                                              std::move(tags)));
   }
+  HistogramSharedPtr hist_tls_ptr;
+  hist_tls_ptr.reset(new ThreadLocalHistogramImpl(final_name, parent_,
```

Comment:
nit: merge into previous line, or do directly in the context of addTlsHistogram()